extensions/nns/src/install_nns.rs

```diff
@@ -416,38 +412,24 @@ pub struct IcNnsInitOpts {
 ///   - This won't work with an HSM, because the agent holds a session open
 ///   - The provider_url is what the agent connects to, and forwards to the replica.
 #[context("Failed to install NNS components.")]
-pub async fn ic_nns_init(ic_nns_init_path: &Path, opts: &IcNnsInitOpts) -> anyhow::Result<()> {
-    let mut cmd = std::process::Command::new(ic_nns_init_path);
-    cmd.arg("--pass-specified-id");
-    cmd.arg("--url");
-    cmd.arg(&opts.nns_url);
-    cmd.arg("--wasm-dir");
-    cmd.arg(&opts.wasm_dir);
-    opts.test_accounts.iter().for_each(|account| {
-        cmd.arg("--initialize-ledger-with-test-accounts");
-        cmd.arg(account);
-    });
-    opts.sns_subnets.iter().for_each(|subnet| {
-        cmd.arg("--sns-subnet");
-        cmd.arg(subnet);
-    });
-    let args: Vec<_> = cmd
-        .get_args()
-        .into_iter()
-        .map(OsStr::to_string_lossy)
-        .collect();
-    dbg!(&cmd);
-    println!("ic-nns-init {}", args.join(" "));
-    cmd.stdout(std::process::Stdio::inherit());
-    cmd.stderr(std::process::Stdio::inherit());
-    let output = cmd
-        .output()
-        .with_context(|| format!("Error executing {:#?}", cmd))?;
-
-    if !output.status.success() {
-        return Err(anyhow!("ic-nns-init call failed"));
+pub async fn ic_nns_init(opts: &IcNnsInitOpts) -> anyhow::Result<String> {
+    let mut args = vec![
+        "--pass-specified-id",
+        "--url",
+        &opts.nns_url,
+        "--wasm-dir",
+        opts.wasm_dir.to_str().unwrap(),
```

Comment:
Can you think of a way to do this without a lossy conversion from PathBuf to String?

Reply:
turned everything into OsString